ts/features/messages/hooks/useAttachmentDownload.tsx

```diff
@@ -95,13 +124,17 @@ export const useAttachmentDownload = (
       return;
     }
 
-    const path = pot.toUndefined(downloadPot)?.path;
-    const fileExists = path !== undefined ? await RNFS.exists(path) : false;
-    if (fileExists) {
-      await openAttachment();
-    } else {
-      dispatch(downloadAttachment.request(attachment));
-    }
+    await pipe(
+      downloadPot,
+      pot.toOption,
+      TE.fromOption(() => new Error("The pot is None")),
```

Comment:
If I am not wrong, we are not really processing any left branch so what if we replace every `() => new Error` and `E.toError` with `() => undefined`?